pkg/pillar/cmd/watcher/watcher.go

```diff
@@ -268,11 +529,14 @@ func Run(ps *pubsub.PubSub, loggerArg *logrus.Logger, logArg *base.LogObject, ar
 	// Handle memory pressure events by calling GC explicitly
 	go handleMemoryPressureEvents()
 
+	go goroutinesMonitor(&ctx)
+
 	for {
 		select {
 		case change := <-subGlobalConfig.MsgChan():
 			subGlobalConfig.ProcessChange(change)
 			setForcedGOGCParams(&ctx)
+			readGlobalGoroutineLeakDetectionParams(&ctx)
```

Comment:
Don't you also need this on line 517?

Reply:
I'm thinking now if a more proper place for this is in `handleGlobalConfigImpl`...

Reply:
Yep.

Reply:
Done